### PR TITLE
Update Rating api and stories

### PR DESCRIPTION
**Pull request by tomi-msft**
This PR is to update the Rating api to remove the `compact` and `readonly` props and add the `mode` prop. It also updates the stories and adds the `SPEC.md` file.

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/a0015d1d41f456bf3ff8a8702a79e54791c5b028'>a0015d1d41f456bf3ff8a8702a79e54791c5b028</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-components/react-rating-preview/src/components/Rating/useRating.tsx [hunk 1]
```diff
@@ -23,15 +23,14 @@ import { StarFilled, StarRegular } from '@fluentui/react-icons';
 export const useRating_unstable = (props: RatingProps, ref: React.Ref<HTMLDivElement>): RatingState => {
   const generatedName = useId('rating-');
   const {
-    appearance = props.readOnly ? 'filled' : 'outline',
-    compact = false,
+    appearance = props.mode === 'readonly' ? 'filled' : 'outline',
```
Comment (khmakoto) [inline packages/react-components/react-rating-preview/src/components/Rating/useRating.tsx:26]: Should you be checking for `readonly-compact` as well here?

File: packages/react-components/react-rating-preview/stories/Rating/index.stories.tsx [hunk 2]
```diff
@@ -4,10 +4,11 @@ import descriptionMd from './RatingDescription.md';
 import bestPracticesMd from './RatingBestPractices.md';
 
 export { Default } from './RatingDefault.stories';
-export { Compact } from './RatingCompact.stories';
+//export { Compact } from './RatingCompact.stories';
```
Comment (khmakoto) [inline packages/react-components/react-rating-preview/stories/Rating/index.stories.tsx:7]: _nit:_ Remove this line

```suggestion
```

File: packages/react-components/react-rating-preview/stories/Rating/index.stories.tsx [hunk 3]
```diff
@@ -4,10 +4,11 @@ import descriptionMd from './RatingDescription.md';
...
 export { Max } from './RatingMax.stories';
+export { Mode } from './RatingMode.stories';
 export { Precision } from './RatingPrecision.stories';
-export { ReadOnly } from './RatingReadOnly.stories';
+//export { ReadOnly } from './RatingReadOnly.stories';
```
Comment (khmakoto) [inline packages/react-components/react-rating-preview/stories/Rating/index.stories.tsx:11]: _nit:_ Remove this line

```suggestion
```

File: packages/react-components/react-rating-preview/src/components/Rating/useRating.tsx (hunk 1 above)
Comment (tomi-msft) [reply packages/react-components/react-rating-preview/src/components/Rating/useRating.tsx]: the api surface for this prop has changed

**Review by khmakoto — APPROVED** [review]

File: packages/react-components/react-rating-preview/stories/Rating/RatingStep.stories.tsx [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
+export const Step = () => {
+  return <Rating step={0.5} defaultValue={3.5} />;
 };
 
-Precision.parameters = {
+Step.parameters = {
   docs: {
     description: {
-      story: 'You can specify half values in the Rating with the "precision" prop.',
+      story: 'You can specify half values in the Rating with the "step" prop.',
```
Comment (behowell) [inline packages/react-components/react-rating-preview/stories/Rating/RatingStep.stories.tsx:11]:
```suggestion
      story: 'You can specify half values in the Rating with `step={0.5}`.',
```

File: packages/react-components/react-rating-preview/stories/Rating/RatingColor.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+      <Rating defaultValue={3.5} ratingLabel={3.5} ratingCountLabel="1,160" />
+    </div>
+  );
+};
+
+Color.parameters = {
+  docs: {
+    description: {
+      story: 'You can specify the appearance of the Rating component with the "color" prop.',
```
Comment (behowell) [inline packages/react-components/react-rating-preview/stories/Rating/RatingColor.stories.tsx:29]: nit:
```suggestion
      story: 'A Rating's `color` can be `neutral` (default), `brand`, or `marigold`.',
```

Also probably re-order the examples in the story itself to match that order (neutral, brand, marigold).

**Pull request merged**